### PR TITLE
chore(snownet): assert that we can send ICMP packets through the tunnel

### PR DESCRIPTION
**Pull request by thomaseizinger**
This is extracted out of #4568 to make that PR smaller. Plus, I'd like to use these new assertions in #4615.

### CODE REVIEW

File: rust/connlib/snownet/tests/lib.rs [hunk 1]
```diff
@@ -855,3 +924,90 @@ fn progress(
...
+                },
+                ip::IpNextHeaderProtocols,
+                ipv4::MutableIpv4Packet,
+                MutablePacket as _, Packet as _,
+            };
+
+            let mut buf = vec![0u8; 60];
+
+            let mut ipv4_packet = MutableIpv4Packet::new(&mut buf[..]).unwrap();
+            ipv4_packet.set_version(4);
```
Comment (conectado) [inline rust/connlib/snownet/tests/lib.rs:944]: I found for tests it's a bit easier to read if we use [populate](https://docs.rs/pnet/latest/pnet/packet/ipv4/struct.MutableIpv4Packet.html#method.populate) but I don't have a strong opinion

**Review by conectado — COMMENTED** [review]
lgtm

File: rust/connlib/snownet/tests/lib.rs (hunk 1 above)
Comment (thomaseizinger) [reply rust/connlib/snownet/tests/lib.rs:944]: I want to extract our own `ip-packet` crate first and then we can make this nicer :)

**Comment by thomaseizinger** [comment]
> lgtm

@conectado Can you approve or is something blocking?

**Review by conectado — APPROVED** [review]

**Pull request merged**